server/etcdserver/api/etcdhttp/health.go

```diff
@@ -182,14 +189,195 @@ func checkAPI(lg *zap.Logger, srv ServerHealth, serializable bool) Health {
 	h := Health{Health: "true"}
 	cfg := srv.Config()
 	ctx, cancel := context.WithTimeout(context.Background(), cfg.ReqTimeout())
-	_, err := srv.Range(ctx, &pb.RangeRequest{KeysOnly: true, Limit: 1, Serializable: serializable})
+	err := rangeCheck(srv, serializable)(ctx)
 	cancel()
-	if err != nil && err != auth.ErrUserEmpty && err != auth.ErrPermissionDenied {
+	if err != nil {
 		h.Health = "false"
-		h.Reason = fmt.Sprintf("RANGE ERROR:%s", err)
+		h.Reason = err.Error()
 		lg.Warn("serving /health false; Range fails", zap.Error(err))
 		return h
 	}
 	lg.Debug("serving /health true")
 	return h
 }
+
+type HealthCheck func(ctx context.Context) error
+
+type CheckRegistry struct {
+	path string
+
+	checks map[string]HealthCheck
+	// the list of checkNames are needed to ensure the order of checks.
+	checkNames []string
+}
+
+func installLivezEndpoints(lg *zap.Logger, mux *http.ServeMux, server ServerHealth) {
+	reg := CheckRegistry{path: "/livez", checks: make(map[string]HealthCheck)}
+	reg.register("serializable_read", rangeCheck(server, true))
+	reg.installHttpEndpoints(lg, mux)
+}
+
+func installReayzEndpoints(lg *zap.Logger, mux *http.ServeMux, server ServerHealth) {
+	reg := CheckRegistry{path: "/readyz", checks: make(map[string]HealthCheck)}
+	reg.register("data_corruption", activeAlarmCheck(server, pb.AlarmType_CORRUPT))
+	reg.register("serializable_read", rangeCheck(server, true))
+	reg.installHttpEndpoints(lg, mux)
+}
+
+func (reg *CheckRegistry) register(name string, check HealthCheck) {
+	reg.checkNames = append(reg.checkNames, name)
+	reg.checks[name] = check
+}
+
+func (reg *CheckRegistry) installHttpEndpoints(lg *zap.Logger, mux *http.ServeMux) {
+	lg.Info(fmt.Sprintf("Installing %s http endpoint.", reg.path))
```

Comment:
```suggestion
	lg.Info("Installing http endpoint", "path", reg.path)
```

Reply:
Note, we don't log registration of any other endpoint. I don't see a reason to make an exception add this logging just for health endpoints.